src/ln/channelmanager.rs

```diff
@@ -386,7 +395,7 @@ macro_rules! handle_error {
 			Ok(msg) => Ok(msg),
 			Err(MsgHandleErrInternal { err, shutdown_finish }) => {
 				if let Some((shutdown_res, update_option)) = shutdown_finish {
-					$self.finish_force_close_channel(shutdown_res);
+					$self.finish_force_close_channel(shutdown_res,true);
```

Comment:
Why not put the flag in ShutdownResult so you dont have to keep track of it everywhere?

Reply:
I will push a version that is not like it. But that version will require a send a bool back from channel.force_shutdown() or I have to send it in like it was up onto now?

Reply:
Another option is perhaps is till remove the new error type, and we only save a flag in the channel indicating whenever that error occurred or not.